python/ray/autoscaler/autoscaler.py

```diff
@@ -389,6 +390,19 @@ def recover_if_needed(self, node_id, now):
         updater.start()
         self.updaters[node_id] = updater
 
+    def _get_commands(self, node_id: str, commands_key: str):
+        commands = self.config[commands_key]
+        node_tags = self.provider.node_tags(node_id)
+        if TAG_RAY_USER_NODE_TYPE in node_tags:
+            node_type = node_tags[TAG_RAY_USER_NODE_TYPE]
+            assert node_type in self.available_node_types,\
+                "Unknown node type tag: {}.".format(
```

Comment:
Don't assert, raise ValueError instead. Asserts should be for internal logic asserts and not checking user input.